### PR TITLE
feat(UI): Add vault configuration page

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds the new Vault configuration page.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/install/php/Update-24.09.0.php [hunk 1]
```diff
@@ -208,11 +208,30 @@
     );
 };
 
+// Vault configuration
```
Comment (a-launois) [inline centreon/www/install/php/Update-24.09.0.php:211]: This should be in Update-24.10.0.php

**Review by a-launois — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by Noha-ElAbrouki — DISMISSED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "L'addresse est invalide"
+
+msgid "Vault address"
+msgstr "Addresse du Vault"
+
+msgid "Root path"
+msgstr "Chemin de la racine"
+
+msgid "Role ID"
+msgstr "Identité du Role"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19436]:
```suggestion
msgstr "Role ID"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+
+msgid "Secret ID"
+msgstr "Identité du secret"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19439]:
```suggestion
msgstr "Secret ID"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "Le formulaire sera réinitialisé et les données modifiées seront perdues. Souhaitez-vous continuer ?"
+
+msgid "Port expected at most 65535"
+msgstr "Le port maximum attendu est 65535"
+
+msgid "Vault configuration updated"
+msgstr "La configuration du Vaul a été mise à jour"
+
+msgid "Post must start from 1"
+msgstr "Le port doit commencé à 1"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19454]:
```suggestion
msgstr "Le numéro de port doit être au moins 1"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "Réinitialiser la configuration"
+
+msgid "The form will be cleared and all data will be lost. Do you want to proceed?"
+msgstr "Le formulaire sera réinitialisé et les données modifiées seront perdues. Souhaitez-vous continuer ?"
+
+msgid "Port expected at most 65535"
+msgstr "Le port maximum attendu est 65535"
+
+msgid "Vault configuration updated"
+msgstr "La configuration du Vaul a été mise à jour"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19451]:
```suggestion
msgstr "La configuration du Vault a été mise à jour"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "Le port doit commencé à 1"
+
+msgid "Migrate"
+msgstr "Migrer"
+
+msgid "Migration script"
+msgstr "Script de migration"
+
+msgid "Execute this script as root on the central."
+msgstr "Script à exécuter en tant que root sur le central."
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19463]:
```suggestion
msgstr "Exécuter ce script en tant que root sur le serveur central."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+
+msgid "The following script will migrate all your credentials from Centreon, Auto-discovery and the database."
+msgstr "Le script suivant migrera vos identifiances de Centreon, Auto-discovery ainsi que de la base de données."
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19466]:
```suggestion
msgstr "Le script suivant migrera vos identifiants de Centreon, d'Auto-discovery ainsi que de la base de données."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+
+msgid "By executing this script, your platform will be in a locked mode and you not be able to do <strong>anything</strong> during the migration process (example: exporting your configuration, updating administration parameters, etc...)."
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19468]:
```suggestion
msgid "By executing this script, your platform will be in a locked mode and you will not be able to do <strong>anything</strong> during the migration process (example: exporting your configuration, updating administration parameters, etc...)."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "En exécutant ce script, votre plateforme entrera sera vérouillé and vous ne pourrez plus <strong>rien modifier</strong> durant le processus de migration (exemple: exporter la configuration, mettre les paramètres d'administration, etc...)."
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19469]:
```suggestion
msgstr "En exécutant ce script, votre plateforme sera verrouillée and vous ne pourrez plus <strong>rien modifier</strong> durant le processus de migration (exemple: déployer la configuration, modifier les paramètres d'administration, etc...)."
```

File: centreon/www/front_src/src/VaultConfiguration/VaultConfiguration.cypress.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+      .clear()
+      .type('http://example.com', { delay: 0 });
+    cy.findByLabelText(labelVaultAddress).blur();
+
+    cy.findByLabelText(labelVaultAddress).should('have.value', 'example.com');
+
+    cy.makeSnapshot();
+  });
+
+  it('sanitizes URL and port field when URL field receives an URL and a port', () => {
```
Comment (cg-tw) [inline centreon/www/front_src/src/VaultConfiguration/VaultConfiguration.cypress.spec.tsx:121]:
```suggestion
  it('sanitizes the URL field when it receives a URL and a port', () => {
```

File: centreon/www/front_src/src/VaultConfiguration/VaultConfiguration.cypress.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    initialize();
+
+    cy.waitForRequest('@getVaultConfiguration');
+
+    cy.contains(labelMigrate).click();
+
+    cy.contains(labelMigrationScript).should('be.visible');
+    cy.contains(labelMigrationScriptExportCredentials).should('be.visible');
+    cy.contains(
+      'By executing this script, your platform will be in a locked mode and you not be able to do'
```
Comment (cg-tw) [inline centreon/www/front_src/src/VaultConfiguration/VaultConfiguration.cypress.spec.tsx:225]:
```suggestion
      'By executing this script, your platform will be in a locked mode and you will not be able to do'
```

File: centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+export const labelSecretID = 'Secret ID';
+export const labelReset = 'Reset';
+export const labelSave = 'Save';
+export const labelCancel = 'Cancel';
+export const labelResetConfiguration = 'Reset configuration';
+export const labelFormWillBeCleared =
+  'The form will be cleared and all data will be lost. Do you want to proceed?';
+export const labelPortExpectedAtMost = 'Port expected at most 65535';
+export const labelVaultConfigurationUpdate = 'Vault configuration updated';
+export const labelPortMustStartFrom1 = 'Port must start from 1';
```
Comment (cg-tw) [inline centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts:17]:
```suggestion
export const labelPortMustStartFrom1 = 'Port number must be at least 1';
```

File: centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+export const labelRootPath = 'Root path';
+export const labelRoleID = 'Role ID';
+export const labelSecretID = 'Secret ID';
+export const labelReset = 'Reset';
+export const labelSave = 'Save';
+export const labelCancel = 'Cancel';
+export const labelResetConfiguration = 'Reset configuration';
+export const labelFormWillBeCleared =
+  'The form will be cleared and all data will be lost. Do you want to proceed?';
+export const labelPortExpectedAtMost = 'Port expected at most 65535';
```
Comment (cg-tw) [inline centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts:15]:
```suggestion
export const labelPortExpectedAtMost = 'Port number must be at most 65535';
```

File: centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+export const labelVaultConfigurationUpdate = 'Vault configuration updated';
+export const labelPortMustStartFrom1 = 'Port must start from 1';
+export const labelMigrate = 'Migrate';
+export const labelMigrationScript = 'Migration script';
+export const labelExecuteThisCommandAsRoot =
+  'Execute this script as root on the central.';
+export const labelMigrationScriptExportCredentials =
+  'The following script will migrate all your credentials from Centreon, Auto-discovery and the database.';
+export const labelByExecutingThisScript =
+  'By executing this script, your platform will be in a locked mode and you not be able to do <strong>anything</strong> during the migration process (example: exporting your configuration, updating administration parameters, etc...).';
```
Comment (cg-tw) [inline centreon/www/front_src/src/VaultConfiguration/translatedLabels.ts:25]:
```suggestion
  'By executing this script, your platform will be in a locked mode and you will not be able to do <strong>anything</strong> during the migration process (example: deploying your configuration, updating administration parameters, etc...).';
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -19419,3 +19419,55 @@ msgstr "Afficher la date"
...
+msgstr "Identité du secret"
+
+msgid "Reset configuration"
+msgstr "Réinitialiser la configuration"
+
+msgid "The form will be cleared and all data will be lost. Do you want to proceed?"
+msgstr "Le formulaire sera réinitialisé et les données modifiées seront perdues. Souhaitez-vous continuer ?"
+
+msgid "Port expected at most 65535"
+msgstr "Le port maximum attendu est 65535"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19448]:
```suggestion
msgstr "Le numéro de port maximum attendu est 65535"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 8 above)
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19468]:
```suggestion
msgid "By executing this script, your platform will be in a locked mode and you not be able to do <strong>anything</strong> during the migration process (example: deploying your configuration, updating administration parameters, etc...)."
```

**Review by cg-tw — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by cg-tw — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by Noha-ElAbrouki — DISMISSED** [review]

File: centreon/www/front_src/src/VaultConfiguration/Form/MigrationModal.tsx [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+        <Typography>{t(labelMigrationScriptExportCredentials)}</Typography>
+        <Typography>
+          <Trans t={t}>{labelByExecutingThisScript}</Trans>
+        </Typography>
+        <Typography>{t(labelMigrationCanTakeSeveralMinutes)}</Typography>
+        <CopyCommand
+          text={`# ${t(labelExecuteThisCommandAsRoot)}
+/usr/share/centreon/bin/migrateCredentials.php`}
+          language="bash"
+          commandToCopy="/usr/share/centreon/bin/migrateCredentials.php"
```
Comment (jeremyjaouen) [inline centreon/www/front_src/src/VaultConfiguration/Form/MigrationModal.tsx:33]:
```suggestion
          commandToCopy="php /usr/share/centreon/bin/migrateCredentials.php"
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by Noha-ElAbrouki — APPROVED** [review]

**Review by Med1233 — APPROVED** [review]

**Pull request merged**